drivers/auxiliary/myDewControllerPro.h

```diff
@@ -0,0 +1,265 @@
+#define MDCP_GET_DEW_POINT "D#"
+#define MDCP_GET_DEW_POINT_RESPONSE "D%f$"
+
+
+//Temperature Probe Commands
+
+#define MDCP_GET_PROBE_TEMPS "C#"
+#define MDCP_SET_TEMP_CH1_OFFSET "[%.1f#"
+#define MDCP_SET_TEMP_CH2_OFFSET "]%.1f#"
+#define MDCP_SET_TEMP_CH3_OFFSET "%%%.1f#"
+#define MDCP_CLEAR_TEMP_OFFSETS "&#"
+#define MDCP_GET_TEMP_OFFSETS "?#"
+#define MDCP_GET_TEMP_OFFSETS_RESPONSE "?%f#%f#%f$"
+#define MDCP_GET_TEMP_RESPONSE "C%f#%f#%f$"
+
+//Board Cooling Fan Commands
+
+#define MDCP_GET_FAN_SPEED "F#"
+#define MDCP_SET_FAN_SPEED "s%d#"
+#define MDCP_GET_BOARD_TEMP "K#"
+#define MDCP_GET_BOARD_TEMP_RESPONSE "K%f$"
+#define MDCP_GET_FAN_ON_TEMP "J#"
+#define MDCP_GET_FAN_ON_TEMP_RESPONSE "J%d$"
+#define MDCP_GET_FAN_OFF_TEMP "L#"
+#define MDCP_GET_FAN_OFF_TEMP_RESPONSE "L%d$"
+#define MDCP_SET_FAN_ON_TEMP "N%d#"
+#define MDCP_SET_FAN_OFF_TEMP "M%d#"
+#define MDCP_GET_FAN_MODE "O#"
+#define MDCP_GET_FAN_MODE_RESPONSE "O%d$"
+#define MDCP_SET_FAN_MODE "I%d#"
+
+//Dew Strap Commands
+
+#define MDCP_BOOST_CH1 "1#"
+#define MDCP_BOOST_CH2 "2#"
+#define MDCP_CANCEL_BOOST "n#"
+#define MDCP_GET_CHANNEL_POWER "W#"
+#define MDCP_GET_CHANNEL_POWER_RESPONSE "W%d#%d#%d$"
+#define MDCP_GET_CH3_SETTINGS "E#"
+#define MDCP_GET_CH3_SETTINGS_RESPONSE "E%d$"
+#define MDCP_SET_CH3_SETTINGS "S%d#"
+#define MDCP_SET_CH3_MANUAL_POWER "G%d#"
+
+#define MDCP_RES_LEN 80
+
+
+/******************************************************************************/
+
+namespace Connection
+{
+class Serial;
+};
+
+class myDewControllerPro : public INDI::DefaultDevice
+{
+    public:
+        myDewControllerPro();
+        virtual ~myDewControllerPro() = default;
+
+        virtual const char *getDefaultName() override;
+        virtual bool initProperties() override;
+        virtual bool updateProperties() override;
+        virtual bool ISNewNumber(const char *dev, const char *name, double values[], char *names[], int n) override;
+        virtual bool ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n) override;
+        virtual void TimerHit() override;
+
+    private:
+        bool sendCommand(const char *cmd, char *response);
+        bool Handshake();
+        bool Ack();
+        bool readMainValues();
+        bool readLCDDisplayValues();
+        bool readBoardFanValues();
+        bool readOffsetValues();
+        bool setOutputBoost(unsigned int channel);
+        bool cancelOutputBoost();
+        bool setTempCalibrations(float ch1, float ch2, float ch3, int ambient);
+        bool setFanTempTrigger(int tempOn, int tempOff);
+        bool zeroTempCalibrations();
+        bool setInt(int mode, const char* mask, const char* errMessage);
+        bool setChoice(int testInt, const char* positiveChoice, const char* negativeChoice, const char* errMessage);
+
+        Connection::Serial *serialConnection{ nullptr };
+        int PortFD{ -1 };
+
+        INumber OutputsN[3];
+        INumberVectorProperty OutputsNP;
```

Comment:
Please use modern INDI properties, i.e.

INDI::PropertyNumber OutputsNP {3}

Reply:
I started using that but thought it was old not new as most drivers are the way I wrote. Where is drivers.xml located please.